### PR TITLE
add a couple useful methods

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: cid.go [hunk 1]
```diff
@@ -408,6 +409,17 @@ func (c Cid) Bytes() []byte {
...
 }
 
+// ByteLen returns len(c.Bytes()) without an allocation
+func (c Cid) ByteLen() int {
+	return len(c.str)
+}
+
+// WriteTo writes the cids bytes to the given writer
+func (c Cid) WriteTo(w io.Writer) error {
+	_, err := io.WriteString(w, c.str)
```
Comment (willscott) [inline cid.go:419]: worth checking if this has fully written the string?

Comment (whyrusleeping) [reply cid.go:419]: actually, probably better just to satisfy the WriterTo interface and return the write size

Comment (willscott) [reply cid.go]: this won't work with the IO methods unless it has a loop as i'm reading the interface contract.
It should either go until there's an error, or until it's written the full string - right?

Comment (whyrusleeping) [reply cid.go]: mmm, i think youre right there.... Maybe i'll pick a different function name. I don't particularly care for loops here

**Pull request merged**